File: Solutions/1732/1732.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1732.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-11-19 15:11:49
***************************************************************************************************/

int largestAltitude(int* gain, int gainSize){
    int ret=fmax(0, gain[0]);
    for (int i=1; i<gainSize; i++) {
        gain[i] += gain[i-1];
        ret = fmax(ret, gain[i]);
    }
    return ret;
}

File: Solutions/1732/1732.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/11/18 15:12
# File: 1732.py
# Desc: 

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        for i in range(1,len(gain)):gain[i] += gain[i-1]
        return max(max(gain), 0)
